common/modal.js

```diff
@@ -60,16 +60,18 @@
         }
 
     }])
-    .controller('SearchPopupController', ['$scope', '$uibModalInstance', 'DataUtils', 'params', 'Session', function SearchPopupController($scope, $uibModalInstance, DataUtils, params, Session) {
+    .controller('SearchPopupController', ['$scope', '$uibModalInstance', 'DataUtils', 'params', 'Session','modalBox', function SearchPopupController($scope, $uibModalInstance, DataUtils, params, Session, modalBox) {
         var vm = this;
 
         vm.params = params;
         vm.ok = ok;
         vm.cancel = cancel;
+        vm.submit = submitMutliSelection;
 
         var reference = params.reference;
         vm.hasLoaded = false;
         var reference = vm.reference = params.reference;
+        var selectMode = params.selectMode?params.selectMode:modalBox.singleSelectMode;
```

Comment:
Again, we should make sure this property is required for every popup select modal rather than having conditional logic for it.

We don't need to worry about backwards compatibility for this feature because only we are leveraging it and can make the proper changes in our current codebase to account for it being required.